src/screens/Wallet.js

```diff
@@ -19,522 +19,505 @@ import WalletAddress from '../components/WalletAddress';
-
-
-const mapDispatchToProps = dispatch => {
-  return {
-    updateWords: (data) => dispatch(updateWords(data)),
-    getHistory: (tokenId) => dispatch(tokenFetchHistoryRequested(tokenId)),
-    getBalance: (tokenId) => dispatch(tokenFetchBalanceRequested(tokenId)),
-  };
-};
-
-const mapStateToProps = (state) => {
-  return {
-    selectedToken: state.selectedToken,
-    tokensHistory: state.tokensHistory,
-    tokensBalance: state.tokensBalance,
-    tokenMetadata: state.tokenMetadata || {},
-    tokens: state.tokens,
-    wallet: state.wallet,
-    walletState: state.walletState,
-    useWalletService: state.useWalletService,
-  };
-};
+import { useHistory } from "react-router-dom";
 
 
 /**
  * Main screen of the wallet
  *
  * @memberof Screens
  */
-class Wallet extends React.Component {
-  static contextType = GlobalModalContext;
-
-  constructor(props) {
-    super(props);
-
-    this.alertSuccessRef = React.createRef();
-  }
-
-  /**
-   * backupDone {boolean} if words backup was already done
-   * successMessage {string} Message to be shown on alert success
-   * shouldShowAdministrativeTab {boolean} If we should display the Administrative Tools tab
-   */
-  state = {
-    backupDone: true,
-    successMessage: '',
-    hasTokenSignature: false,
-    shouldShowAdministrativeTab: false,
-    totalSupply: null,
-    canMint: false,
-    canMelt: false,
-    transactionsCount: null,
-    mintCount: null,
-    meltCount: null,
-  };
-
-  // Reference for the unregister confirm modal
-  unregisterModalRef = React.createRef();
-
-  componentDidMount = async () => {
-    this.setState({
-      backupDone: LOCAL_STORE.isBackupDone()
-    });
-
-    this.initializeWalletScreen();
-  }
-
-  componentDidUpdate(prevProps) {
-    // the selected token changed, we should re-initialize the screen
-    if (this.props.selectedToken !== prevProps.selectedToken) {
-      this.initializeWalletScreen();
-    }
-
-    // if the new selected token history changed, we should fetch the token details again
-    const prevTokenHistory = get(prevProps.tokensHistory, this.props.selectedToken, {
-      status: TOKEN_DOWNLOAD_STATUS.LOADING,
-      updatedAt: -1,
-      data: [],
-    });
-    const currentTokenHistory = get(this.props.tokensHistory, this.props.selectedToken, {
+function Wallet() {
+  // Modal context
+  const context = useContext(GlobalModalContext);
+
+  // State
+  /** backupDone {boolean} if words backup was already done */
+  const [backupDone, setBackupDone] = useState(LOCAL_STORE.isBackupDone());
+  /** successMessage {string} Message to be shown on alert success */
+  const [successMessage, setSuccessMessage] = useState('');
+  /* shouldShowAdministrativeTab {boolean} If we should display the Administrative Tools tab */
+  const [shouldShowAdministrativeTab, setShouldShowAdministrativeTab] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(''); // TODO: Metadata token error are being suppressed as of now
```

Comment:
Even though issue #487 is already open to fix this I found it best to leave it as a warning in the code itself, because this is a bug.

Do you agree with leaving it as a comment?

Reply:
I mean, `setErrorMessage` is not being used anywhere in this file, and it's ageneric error message

So why keep it there?

Maybe remove the `const [errorMessage, setErrorMessage] = useState('');` part and update the comment to explain that metadata fetch errors are being ignored

Reply:
Done on [8098579](https://github.com/HathorNetwork/hathor-wallet/pull/488/commits/8098579ce2419fbdc73223ab0442d0842b6f5d70)